chore(release): version 2.3.0

Mednoob opened the pull request:
Will be merged at the end of 2021. Feel free to open pull request to the branch ([`feat/2.3.0`](https://github.com/zhycorp/disc-11/tree/feat/2.3.0)).

Commit:
feat(queue): Fetch Spotify data from API instead of scraping (#634)

Commit:
chore(env_example): tidy up

dependabot[bot] committed:
chore(deps-dev): bump eslint from 8.3.0 to 8.4.0 (#636)

Bumps [eslint](https://github.com/eslint/eslint) from 8.3.0 to 8.4.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.3.0...v8.4.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Mednoob committed:
feat(typings): add 'ISportifyAccessTokenAPIResult' interface

Mednoob committed:
feat(typings): make accessToken property to optional

Mednoob committed:
feat(SpotifyUtils): do some types and eslint adjustment

Mednoob committed:
style(typings): change from 'Array<OBJECT>' to 'OBJECT[]'

Commit:
Merge branch 'main' into feat/2.3.0

mzrtamp reviewed (approved):
Approved for now, waiting for some known bugs fixes

dependabot[bot] committed:
chore(deps): bump youtube-dl-exec from 2.0.0 to 2.0.1 (#644)

Bumps [youtube-dl-exec](https://github.com/microlinkhq/youtube-dl-exec) from 2.0.0 to 2.0.1.
- [Release notes](https://github.com/microlinkhq/youtube-dl-exec/releases)
- [Changelog](https://github.com/microlinkhq/youtube-dl-exec/blob/master/CHANGELOG.md)
- [Commits](https://github.com/microlinkhq/youtube-dl-exec/compare/v2.0.0...v2.0.1)

---
updated-dependencies:
- dependency-name: youtube-dl-exec
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Mednoob committed:
feat(CommandContext): also edit reply if the context is deferred

Mednoob committed:
chore(deps): update lockfile

Mednoob committed:
feat(commands/music): create remove command

Mednoob committed:
fix(RemoveCommand): forgot to save file and commit change

dependabot[bot] committed:
chore(deps): bump got from 11.8.3 to 12.0.0 (#645)

Bumps [got](https://github.com/sindresorhus/got) from 11.8.3 to 12.0.0.
- [Release notes](https://github.com/sindresorhus/got/releases)
- [Commits](https://github.com/sindresorhus/got/compare/v11.8.3...v12.0.0)

---
updated-dependencies:
- dependency-name: got
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/eslint-plugin (#648)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.6.0 to 5.7.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.7.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @types/i18n from 0.13.1 to 0.13.2 (#653)

Bumps [@types/i18n](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/i18n) from 0.13.1 to 0.13.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/i18n)

---
updated-dependencies:
- dependency-name: "@types/i18n"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump youtube-dl-exec from 2.0.0 to 2.0.2 (#647)

Bumps [youtube-dl-exec](https://github.com/microlinkhq/youtube-dl-exec) from 2.0.0 to 2.0.2.
- [Release notes](https://github.com/microlinkhq/youtube-dl-exec/releases)
- [Changelog](https://github.com/microlinkhq/youtube-dl-exec/blob/master/CHANGELOG.md)
- [Commits](https://github.com/microlinkhq/youtube-dl-exec/compare/v2.0.0...v2.0.2)

---
updated-dependencies:
- dependency-name: youtube-dl-exec
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/parser from 5.6.0 to 5.7.0 (#649)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.6.0 to 5.7.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.7.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 16.11.12 to 17.0.0 (#651)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.12 to 17.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

Commit:
chore(deps): debump got from 12.0.0 to 11.8.3

Commit:
chore(RemoveCommand): forget to import i18n

Commit:
fix(GeneralUtil): fix already sent or deferred interaction

Mednoob committed:
chore(deps): update lockfile

Mednoob committed:
fix: fix interaction failed on some commands

Mednoob committed:
feat(YTDLUtil): filtering to opus audio codec is actually not needed

Mednoob committed:
chore(lang/en): fix wrong localization naming

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 16.11.12 to 17.0.1 (#655)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.12 to 17.0.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump eslint from 8.4.1 to 8.5.0 (#656)

Bumps [eslint](https://github.com/eslint/eslint) from 8.4.1 to 8.5.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.4.1...v8.5.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Mednoob committed:
feat(GeneralUtil): change spotify result sorting

Mednoob committed:
feat(GeneralUtil): set the result type once

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/eslint-plugin (#663)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.6.0 to 5.8.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.8.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/parser from 5.6.0 to 5.8.0 (#664)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.6.0 to 5.8.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.8.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 16.11.12 to 17.0.2 (#665)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.12 to 17.0.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 16.11.12 to 17.0.4 (#669)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.12 to 17.0.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @types/node from 16.11.12 to 17.0.5 (#670)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.12 to 17.0.5.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

Mednoob committed:
feat(utils): add OperationManager util

Mednoob committed:
chore(lang/en): removing 'voteMessage' from skip command localization

Mednoob committed:
fix(SkipCommand): fix a voting bug

Mednoob committed:
fix(OperationManager): fix linter warning

Mednoob committed:
fix(CommandManager): import i18n from config

Mednoob committed:
feat(ClientUtils): parse FFmpeg version so it only shows number version

Mednoob committed:
chore(deps): update lockfile

Mednoob committed:
style(CommandManager): move i18n import declaration

Mednoob committed:
chore(deps): bump discord.js to 13.5.0

Mednoob committed:
chore(deps-dev): add typescript

Mednoob committed:
refactor: change code after discord.js update

Mednoob committed:
style(env_example): better view of env for glitch users (welp, probably)

dependabot[bot] committed:
ci(deps): bump actions/setup-node from 2.5.0 to 2.5.1 (#675)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.5.0 to 2.5.1.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.5.0...v2.5.1)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump i18n from 0.13.3 to 0.13.4 (#677)

Bumps [i18n](https://github.com/mashpie/i18n-node) from 0.13.3 to 0.13.4.
- [Release notes](https://github.com/mashpie/i18n-node/releases)
- [Changelog](https://github.com/mashpie/i18n-node/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mashpie/i18n-node/compare/0.13.3...0.13.4)

---
updated-dependencies:
- dependency-name: i18n
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump date-fns from 2.27.0 to 2.28.0 (#674)

Bumps [date-fns](https://github.com/date-fns/date-fns) from 2.27.0 to 2.28.0.
- [Release notes](https://github.com/date-fns/date-fns/releases)
- [Changelog](https://github.com/date-fns/date-fns/blob/master/CHANGELOG.md)
- [Commits](https://github.com/date-fns/date-fns/compare/v2.27.0...v2.28.0)

---
updated-dependencies:
- dependency-name: date-fns
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/parser from 5.6.0 to 5.8.1 (#673)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.6.0 to 5.8.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.8.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

dependabot[bot] committed:
chore(deps-dev): bump @typescript-eslint/eslint-plugin (#672)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.6.0 to 5.8.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.8.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mzrtamp <mozartrafi@zhycorp.net>

Commit:
chore(deps): update lockfile

mzrtamp commented on lang/en.json:
```diff
@@ -170,6 +170,15 @@
+				"slashPositionsDescription": "Position of the songs to remove separated by comma or space",
+				"usage": "{prefix}remove <positions>",
+				"noPermission": "You don't have permission to use this command",
+				"noPositions": "Please, specify the song positions to remove.",
```
```suggestion
				"noPositions": "Please specify the song positions to remove.",
```

mzrtamp commented on lang/en.json:
```diff
@@ -170,6 +170,15 @@
+				"description": "Remove song(s) from the queue",
+				"slashPositionsDescription": "Position of the songs to remove separated by comma or space",
+				"usage": "{prefix}remove <positions>",
+				"noPermission": "You don't have permission to use this command",
```
```suggestion
				"noPermission": "You don't have permission to use this command.",
```

mzrtamp commented on lang/en.json:
```diff
@@ -170,6 +170,15 @@
+				"noPermission": "You don't have permission to use this command",
+				"noPositions": "Please, specify the song positions to remove.",
+				"songsRemoved": "Removed {removed} songs from the queue.",
+				"songSkip": "One of the removed song is the now-playing song. Song skipped."
```
```suggestion
				"songSkip": "One of the removed song is the now-playing song, song skipped."
```

mzrtamp reviewed (changes requested):
I'll do it myself

Commit:
chore(lang/en): update string

Commit:
chore(lang/en): update string

Commit:
chore(lang/en): update string

Commit:
chore(release): version 2.3.0

mzrtamp reviewed (approved):
Nice
